### PR TITLE
Fix: do not shuffle collators before applying priority

### PR DESCRIPTION
**Pull request by tmpolaczyk**
The implementation of collator priority in #471 has a bug that renders it useless: collators are being shuffled twice. This resulted in the priority being applied to a shuffled list of collators, so it was equivalent to having no priority.

This PR fixes that by removing one line: `collators.shuffle(&mut rng);`, and fixing all the tests that break because we changed the way we use the randomness. Also adds a new test `assign_collators_truncates_before_shuffling` to check that collator priority works as expected.

### CODE REVIEW

**Review by girazoki — APPROVED** [review]

File: pallets/collator-assignment/src/tests.rs [hunk 1]
```diff
@@ -794,15 +794,15 @@ fn assign_collators_rotation_container_chains_are_shuffled() {
 
         MockData::mutate(|m| {
             // Seed chosen manually to see the case where container 1002 is given priority
-            m.random_seed = [2; 32];
```
Comment (ParthDesai) [inline pallets/collator-assignment/src/tests.rs:797]: Why did you change the random seed? Even if you don't change it, wouldn't it result in different order in fixed code now?

**Review by ParthDesai — APPROVED** [review]
Looks good to me. Just one minor question.

File: pallets/collator-assignment/src/tests.rs (hunk 1 above)
Comment (tmpolaczyk) [reply pallets/collator-assignment/src/tests.rs:797]: If I don't change it collators get assigned to container 1001, and this test is supposed to check the case where they are assigned to 1002.
